tests/Areas/AppConfig/LiveTests/AppConfigCommandTests.cs

```diff
@@ -334,4 +334,330 @@ public async Task Should_set_and_delete_appconfig_kv()
         var keyProperty = result.AssertProperty("key");
         Assert.Equal(key, keyProperty.GetString());
     }
+
+    [Fact]
+    [Trait("Category", "Live")]
+    public async Task Should_set_and_get_appconfig_kv_with_content_type()
+    {
+        // arrange
+        const string key = "config-with-content-type";
+        const string value = @"{""property"":""value""}";
+        const string contentType = "application/json";
+
+        // act - set key-value with content type
+        var setResult = await CallToolAsync(
+            "azmcp-appconfig-kv-set",
+            new()
+            {
+                { "subscription", _subscriptionId },
+                { "account-name", _accountName },
+                { "key", key },
+                { "value", value },
+                { "content-type", contentType }
+            });
+
+        // assert - verify the set result
+        var valueRead = setResult.AssertProperty("value");
+        Assert.Equal(value, valueRead.GetString());
+
+        var contentTypeRead = setResult.AssertProperty("contentType");
+        Assert.Equal(JsonValueKind.String, contentTypeRead.ValueKind);
+        Assert.Equal(contentType, contentTypeRead.GetString());
+
+        // act - get the key-value to verify content type was stored
+        var getResult = await CallToolAsync(
+            "azmcp-appconfig-kv-show",
+            new()
+            {
+                { "subscription", _subscriptionId },
+                { "account-name", _accountName },
+                { "key", key }
+            });
+
+        // assert - verify the get result
+        var setting = getResult.AssertProperty("setting");
+        Assert.Equal(JsonValueKind.Object, setting.ValueKind);
+
+        valueRead = setting.AssertProperty("value");
+        Assert.Equal(JsonValueKind.String, valueRead.ValueKind);
+        Assert.Equal(value, valueRead.GetString());
+
+        contentTypeRead = setting.AssertProperty("contentType");
+        Assert.Equal(JsonValueKind.String, contentTypeRead.ValueKind);
+        Assert.Equal(contentType, contentTypeRead.GetString());
+
+        // cleanup
```

Comment:
nit; We don't need to worry a about cleanup, we use ephemeral live test resources :)